### PR TITLE
refactor Dockerfile & docs

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR attempts to refactor the `Dockerfile`.

Main thing is to avoid having `curl` in the final image by fetching dependencies in a separate build stage and cleanup of separate `ENV` lines. Finally, a retagged python image is used.

I also started adding some docs regarding development (more later)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,70 +1,79 @@
...
+    && mkdir -p /binaries \
+    && curl -SL https://storage.googleapis.com/kubernetes-release/release/v${KUBECTL_VER}/bin/linux/amd64/kubectl -o /binaries/kubectl \
+    && curl -SL https://get.helm.sh/helm-v${HELM_VER}-linux-amd64.tar.gz | \
+       tar -C /binaries --strip-components 1 -xvzf - linux-amd64/helm \
+    && curl -SL https://github.com/giantswarm/apptestctl/releases/download/v${APPTESTCTL_VER}/apptestctl-v${APPTESTCTL_VER}-linux-amd64.tar.gz | \
+       tar -C /binaries --strip-components 1 -xvzf - apptestctl-v${APPTESTCTL_VER}-linux-amd64/apptestctl \
+    && curl -SL https://github.com/helm/chart-testing/releases/download/v${CT_VER}/chart-testing_${CT_VER}_linux_amd64.tar.gz | \
+       tar -C /binaries -xvzf - ct etc/lintconf.yaml etc/chart_schema.yaml && mv /binaries/etc /etc/ct
+
+COPY container-entrypoint.sh /binaries
```
Comment (piontec) [inline Dockerfile:18]: Is it not too early to include it? This file is still part of the code base and its change will trigger quite ineffective build.

**Review by piontec — APPROVED** [review]
Looks all OK, just... `curl` is <1MB, that was quite some effort to shave it :D

**Comment by ubergesundheit** [comment]
> Looks all OK, just... `curl` is <1MB, that was quite some effort to shave it :D

You go me ;) I also wanted to merge the multiple `ENV` lines and add some empty lines to make the file (for me) more readable.

**Comment by piontec** [comment]
> > Looks all OK, just... `curl` is <1MB, that was quite some effort to shave it :D
> 
> You go me ;) I also wanted to merge the multiple `ENV` lines and add some empty lines to make the file (for me) more readable.

Yes, readability is better now :+1:

**Pull request merged**